client/editor/tool_tile.cpp

```diff
@@ -86,17 +137,224 @@ void editor_tool_tile::close_tool()
 }
 
 /**
- * \brief Update the editor tool widget UI elements
+ * \brief Update the editor tool widget UI elements based on the tile
+ * selected
  */
 void editor_tool_tile::update_ett(struct tile *ptile)
 {
-  if (ptile) {
+  if (ptile != nullptr) {
+    clear_hover_state();
     editor_set_current_tile(ptile);
+
+    // tile terrain w/ image
+    ui.value_terrain->setText(
+        QString(terrain_name_translation(tile_terrain(ptile))));
+    struct terrain *pterrain = terrain_by_translated_name(
+        terrain_name_translation(tile_terrain(ptile)));
+    QPixmap pm = get_tile_sprites(ptile, pterrain);
+    ui.pixmap_terrain->setPixmap(pm);
+
+    // tile continent number
     ui.value_continent->setNum(ptile->continent);
+
+    // tile owner
+    struct player *owner = tile_owner(ptile);
+    if (owner != nullptr) {
+      ui.value_owner->setText(nation_adjective_for_player(owner));
+    } else {
+      ui.value_owner->setText(_("None"));
+    }
+
+    // tile coordinates
     ui.value_x->setNum(index_to_map_pos_x(ptile->index));
     ui.value_y->setNum(index_to_map_pos_y(ptile->index));
     ui.value_nat_x->setNum(index_to_native_pos_x(ptile->index));
     ui.value_nat_y->setNum(index_to_native_pos_y(ptile->index));
+
+    // tile resource
+    if (ptile->resource) {
+      struct extra_type *res = ptile->resource;
+      ui.value_resource->setText(extra_name_translation(res));
+    } else {
+      ui.value_resource->setText(_("None"));
+    }
+
+    // tile road (highest level)
+    ui.value_road->setText(get_tile_road_name(ptile));
+    // tile infrastructure
+    ui.value_infra->setText(get_tile_infra_name(ptile));
+    // tile base
+    ui.value_base->setText(get_tile_base_name(ptile));
+    // tile hut
+    ui.value_hut->setText(get_tile_hut_name(ptile));
+    // tile nuisance (pollution, fallout)
+    ui.value_nuisance->setText(get_tile_nuisance_name(ptile));
   }
-  clear_hover_state();
+}
+
+/**
+ * \brief Return a string of the final road on a tile
+ */
+const QString editor_tool_tile::get_tile_road_name(struct tile *ptile)
+{
+  QString sroad = nullptr;
+  extra_type_by_cause_iterate(EC_ROAD, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sroad = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (sroad != nullptr) {
```

Comment:
Null strings are only there for compatibility. Equivalent to:
```suggestion
  if (!sroad.isEmpty()) {
```